src/prefect/results.py

```diff
@@ -62,15 +62,17 @@ def DEFAULT_STORAGE_KEY_FN():
 
 
 @sync_compatible
-async def get_default_result_storage() -> ResultStorage:
+async def get_default_result_storage() -> WritableFileSystem:
     """
     Generate a default file system for result storage.
     """
-    return (
-        await Block.load(PREFECT_DEFAULT_RESULT_STORAGE_BLOCK.value())
-        if PREFECT_DEFAULT_RESULT_STORAGE_BLOCK.value() is not None
-        else LocalFileSystem(basepath=PREFECT_LOCAL_STORAGE_PATH.value())
-    )
+    try:
+        return await Block.load(PREFECT_DEFAULT_RESULT_STORAGE_BLOCK.value())
+    except ValueError:
+        block = LocalFileSystem(basepath=PREFECT_LOCAL_STORAGE_PATH.value())
+        default_name = PREFECT_DEFAULT_RESULT_STORAGE_BLOCK.value().split("/")[-1]
+        await block.save(default_name)
```

Comment:
I think the answer is probably "it's too complicated to do otherwise right now", but is there a concrete reason we need to _save_ the block server-side?

Reply:
Yeah, the results mechanics won't save results unless the block is saved because the block document ID is persisted with the artifact representing the result.